<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QDialog" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>961</width>
    <height>617</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>X-keys Generic Demo</string>
  </property>
  <widget class="QPushButton" name="start_button">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>92</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Connect to the first X-keys device on the system</string>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
  </widget>
  <widget class="QPushButton" name="stop_button">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>10</y>
     <width>92</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Disconnect from the X-keys hardware</string>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </widget>
  <widget class="QPushButton" name="quit_button">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>10</y>
     <width>92</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Exit this application</string>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="output_box">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>941</width>
     <height>161</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="clear_button">
   <property name="geometry">
    <rect>
     <x>860</x>
     <y>220</y>
     <width>92</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Clear the contents of the output box above.</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
  </widget>
  <widget class="QPushButton" name="convert_pid1">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>560</y>
     <width>141</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Convert the device to desired PID. See documentation for endpoints available on each PID for product.</string>
   </property>
   <property name="text">
    <string>Convert PID</string>
   </property>
  </widget>
  <widget class="QPushButton" name="timestamp_on">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>510</y>
     <width>121</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Enable timestamp reporting from the device.</string>
   </property>
   <property name="text">
    <string>Time Stamp On</string>
   </property>
  </widget>
  <widget class="QPushButton" name="timestamp_off">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>510</y>
     <width>121</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Disable timestamp reporting from the device.</string>
   </property>
   <property name="text">
    <string>Time Stamp Off</string>
   </property>
  </widget>
  <widget class="QPushButton" name="show_descriptor">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>510</y>
     <width>131</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Print a description of the device in the output box above.</string>
   </property>
   <property name="text">
    <string>Show Descriptor</string>
   </property>
  </widget>
  <widget class="QPushButton" name="generate_report">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>510</y>
     <width>161</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Cause the device to output a data report. This is useful for getting an initial state of the device.</string>
   </property>
   <property name="text">
    <string>Generate Data Report</string>
   </property>
  </widget>
  <widget class="QPushButton" name="keyboard_reflector">
   <property name="geometry">
    <rect>
     <x>251</x>
     <y>250</y>
     <width>161</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>After clicking focus will change to the textbox and Abcd should be typed. Requires keyboard endpoint.</string>
   </property>
   <property name="text">
    <string>Keyboard Reflector</string>
   </property>
  </widget>
  <widget class="QPushButton" name="joystick_reflector">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>280</y>
     <width>161</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Open Game Controllers control panel to accept commands, sends game buttons 1, 2, 3, and 4. Required joystick endpoint.</string>
   </property>
   <property name="text">
    <string>Joystick Reflector</string>
   </property>
  </widget>
  <widget class="QPushButton" name="mouse_reflector">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>310</y>
     <width>161</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Turn Mouse On and while reading data, press first key to behave as left mouse button. Requires mouse endpoint.</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
   <property name="text">
    <string>Mouse Reflector</string>
   </property>
  </widget>
  <widget class="QPushButton" name="unit_id">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>370</y>
     <width>141</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Writes unit id 0-255 into device.</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
   <property name="text">
    <string>Set Unit ID</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>570</y>
     <width>411</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>0, 1, 2, 3, etc. refer to documentation for specific product.</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>230</y>
     <width>181</width>
     <height>181</height>
    </rect>
   </property>
   <property name="title">
    <string>LED Controls</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <widget class="QCheckBox" name="flash_green_check">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Flash the Green LED.</string>
    </property>
    <property name="text">
     <string>Flash Green</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="green_check">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Turn on the Green LED.</string>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="red_check">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Turn on the Red LED.</string>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="flash_red_check">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Flash the Red LED.</string>
    </property>
    <property name="text">
     <string>Flash Red</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>59</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Freq</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="flash_freq">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>150</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>The frequency to flash the LEDs and backlight at. This is really a period. Low numbers flash faster; high numbers flash slower.</string>
    </property>
    <property name="text">
     <string>50</string>
    </property>
   </widget>
  </widget>
  <widget class="QLineEdit" name="unitid_number">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>370</y>
     <width>61</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>integer 0-255</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
  </widget>
  <widget class="QLabel" name="mouse_label">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>320</y>
     <width>91</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Mouse Off</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>430</y>
     <width>81</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>LED index</string>
   </property>
  </widget>
  <widget class="QPushButton" name="individual_led">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>440</y>
     <width>141</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Control individual LED, see documentation for index values for each product.</string>
   </property>
   <property name="text">
    <string>Individual LED</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="kbd_input">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>250</y>
     <width>61</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>integer 0-255</string>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLineEdit" name="pid_number">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>560</y>
     <width>61</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>integer 0-255</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="led_index">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>420</y>
     <width>61</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>integer 0-255</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="led_state">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>450</y>
     <width>61</width>
     <height>27</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>integer 0-255</string>
   </property>
   <property name="text">
    <string>0</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>460</y>
     <width>241</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>LED state (0=off, 1=on, 2=flash)</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
